src/lib.rs

```diff
@@ -173,19 +173,54 @@ impl Event {
     }
 }
 
+#[cfg(test)]
+mod test_setup {
+    use ctor::ctor;
+    use dotenv::dotenv;
+
+    #[ctor]
+    fn load_dotenv() {
+        dotenv().ok(); // Load the .env file
+        println!("Loaded .env for tests");
+    }
+}
+
 #[cfg(test)]
 pub mod tests {
     use super::*;
-    use chrono::Utc;
+
+    // see https://us.posthog.com/project/115809/ for the e2e project
 
+    #[test]
+    fn inner_event_adds_lib_properties_correctly() {
+        // Arrange
+        let mut event = Event::new("unit test event", "1234");
+        event.insert_prop("key1", "value1").unwrap();
+        let api_key = "test_api_key".to_string();
+
+        // Act
+        let inner_event = InnerEvent::new(event, api_key);
+
+        // Assert
+        let props = &inner_event.properties.props;
+        assert_eq!(
+            props.get("$lib_name"),
+            Some(&serde_json::Value::String("posthog-rs".to_string()))
+        );
+    }
+
+    #[cfg(feature = "e2e-test")]
     #[test]
     fn get_client() {
-        let client = crate::client(env!("POSTHOG_API_KEY"));
+        use std::collections::HashMap;
+
+        let api_key = std::env::var("POSTHOG_RS_E2E_TEST_API_KEY").unwrap();
```

Comment:
Can you add an `.env.example` file to make it clear that we need this to run tests locally? You could also add a section to the README.md explaining that one needs to grab an API key from their instance to be able to run tests